File: 1817-calculate-money-in-leetcode-bank/calculate-money-in-leetcode-bank.cpp
class Solution {
public:
    int totalMoney(int n) {
        int mp[7] = {0,1,3,6,10,15,21};
        int ans = 0, tmp = 0, itr= n/7;
        
        for(int i=0; i<itr; i++){
            ans += 28 + tmp;
            tmp += 7;
        }

        ans += mp[(n%7)] + ((n%7) * tmp/7);

        return ans;
    }
};

File: 1817-calculate-money-in-leetcode-bank/calculate-money-in-leetcode-bank.java
class Solution {
    public int totalMoney(int n) {
        int i=8, ans = 0, sum = 28;
        Queue<Integer> q = new LinkedList<>();

        for(int j=0; j<7; j++) q.add(j+1);

        while(n > 7){
            ans += sum;
            // System.out.println(q + " " + q.peek());
            sum = sum + i - q.poll();
            q.add(i++);
            n -= 7;
        }

        while(n > 0){
            // System.out.println(q.peek());
            ans += q.poll();
            n--;
        }

        return ans;
    }
}